Hymns/Tags/TagStore.swift

```diff
@@ -60,6 +75,21 @@ class TagStoreRealmImpl: TagStore {
             action(favorite)
         }.toNotification()
     }
+
+    // MARK: Realm Query Functions
+
+    /** Can be used either with a value to specificially query for one tag or without the optional to query all tags besides favorites*/
+    func querySelectedTags(tagSelected: String?) -> Results<TagEntity> {
+        guard let specificTag = tagSelected else {
+            return realm.objects(TagEntity.self)
+        }
+        return realm.objects(TagEntity.self).filter(NSPredicate(format: "tags == %@", specificTag))
+    }
```

Comment:
Does this really exclude the Favorites tags?

Reply:
Wouldn't it? Lol because the Favorites tags are in a different entity right? The only problem with doing the PR this way is it's hard for me to check this stuff now lol...

Reply:
If they are all in the subclass FavoriteEntity then those objects would not be stored in the TagsEntity if I am right?

Reply:
Hmm, interesting. I think they still would be, since they're still using the same realm file.

But you can find out pretty easily via a unit test. Create an in-memory database, add stuff to it, and query to see what is and isn't in there: https://github.com/HymnalDreamTeam/Hymns-iOS/blob/master/HymnsTests/History/HistoryStoreRealmImplSpec.swift#L11

Reply:
It looks like I was write. I think you can think of entities in Realm as tables. So they are two different tables. Take a look at my unit test and you will see that indeed favorites are not being stored there. Thanks for the tip to use unit tests like this.

Reply:
okay. Yeah, it looks like realm doesn't really support polymorphism: https://stackoverflow.com/questions/42809394/inheritance-with-swift-realm-confusion.

I think that's okay. Maybe we can just keep it as two tables then. In that case, we won't need a special "*_*_favorites*_*" tag, since they are different tables, and they'll never conflict

Reply:
actually, if we're just going to run with it being a separate table, should we just split it up between FavoritesStore and TagStore? Then things will be cleaner, since we are fully decoupling these into two separate features